app/Actions/Album/Tracks.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Actions\Album;
+
+use App\Models\Album;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+
+class Tracks extends Action
+{
+	/**
+	 * @param string $albumID the ID of the album
+	 * @param mixed  $value   the value to be set
+	 */
+	public function set(string $albumID, UploadedFile $value): void
+	{
+		$album = $this->albumFactory->findAlbumOrFail($albumID, false);
+		if ($album->track_id != null) {
+			Storage::delete("tracks/$album->track_id.xml");
+		}
+
+		$new_track_id = uniqid();
+		Storage::putFileAs('tracks/', $value, "$new_track_id.xml");
+
+		if (Album::query()
+				->where('id', '=', $albumID)
+				->update(['track_id' => $new_track_id]) !== 1
+		) {
+			throw new ModelNotFoundException();
```

Comment:
If you use `$album->save` as recommended as above, explicitly throwing an exception won't be necessary, because `$album->save` already do so.

Moreover, just as a pointer: If you ever need to throw an exception, try to throw one of the Lychee specific exceptions (see `\App\Exceptions\` and `\App\Exceptions\Internal`). If you don't find any which seems to fit, then create a new one.